lib/formatters.js

```diff
@@ -1,27 +1,50 @@
 'use strict';
 
 const Promise = require('bluebird');
+const { DataEntity } = require('@terascope/utils');
 
 // No parsing, leaving to reader or a downstream op.
-function pass(data) {
-    return data;
+function raw(data, logger, metadata, opConfig) {
+    return Promise.map(data, (record) => {
+        try {
+            return DataEntity.make(
+                { data: record },
+                metadata
+            );
+        } catch (err) {
+            if (opConfig._dead_letter_action === 'log') {
+                logger.error('Bad record');
+                logger.error(err);
+            } else if (opConfig._dead_letter_action === 'throw') {
+                throw err;
+            }
+            return null;
+        }
+    });
 }
 
-// Each item parsed as JSON.
-function json(data, logger) {
+function json(data, logger, metadata, opConfig) {
     return Promise.map(data, (record) => {
```

Comment:
Same reason as above

```suggestion
     return data.map(data, (record) => {
```

Reply:
fixed